packages/teamscale-javascript-instrumenter/src/vaccine/buffer.ts

```diff
@@ -36,13 +36,24 @@ export function createCoverageBuffer(flushAfterMillis: number, onFlush: FlushFun
             return fileBuffer;
         }
 
-        fileBuffer = { lines: [] };
+        fileBuffer = { lines: new Set(), };
         buffer.set(fileId, fileBuffer);
         return fileBuffer;
     }
 
-    function putLineCoverage(fileId: string, startLine: number, endLine: number): void {
-        getBufferFor(fileId).lines.push(startLine, endLine);
+    function putLineCoverage(fileId: string, startLine: number, endLine?: number): void {
```

Comment:
Not 100% sure, but this should also work:

```suggestion
    function putLineCoverage(fileId: string, startLine: number, endLine = startLine): void {
```